test/jdk/com/sun/crypto/provider/KDF/HKDFExhaustiveTest.java

```diff
@@ -0,0 +1,552 @@
+    testExtractMethod(hk);
+
+    System.out.println("Testing deriveKey and deriveData with extract method");
+    testDeriveKeyDataWithExtract(hk);
+
+    // Test expand
+    System.out.println("Testing expand method");
+    testExpandMethod(hk);
+
+    System.out.println("Testing deriveKey and deriveData with expand method");
+    testDeriveKeyDataWithExpand(hk);
+
+    // Test ExtractThenExpand
+    System.out.println("Testing extractThenExpand method");
+    testExtractExpandMethod(hk);
+
+    System.out.println("Testing deriveKey and deriveData with extExpand method");
+    testDeriveKeyDataWithExtExpand(hk);
+
+    System.out.println("Test executed successfully.");
+  }
+
+  private static void testGetInstanceMethods()
+      throws InvalidAlgorithmParameterException, NoSuchAlgorithmException, NoSuchProviderException {
+    // POSITIVE TestCase: KDF getInstance methods test
+    for (String algo : KDF_ALGORITHMS) {
+      KdfGetInstanceVerifier.test(algo, SUNJCE, null);
+    }
+  }
+
+  private static void testGetInstanceNegative() {
+    final String INVALID_STRING = "INVALID";
+    final Provider SUNJCE_PROVIDER = Security.getProvider(SUNJCE);
+
+    // getInstance(String algorithm)
+    Utils.runAndCheckException(() -> KDF.getInstance(null), NullPointerException.class);
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(INVALID_STRING), NoSuchAlgorithmException.class);
+
+    // getInstance(String algorithm, String provider)
+    Utils.runAndCheckException(() -> KDF.getInstance(null, SUNJCE), NullPointerException.class);
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(INVALID_STRING, SUNJCE), NoSuchAlgorithmException.class);
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(KDF_ALGORITHMS[0], (String) null), NullPointerException.class);
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(KDF_ALGORITHMS[0], INVALID_STRING), NoSuchProviderException.class);
+
+    // getInstance(String algorithm, Provider provider)
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(null, SUNJCE_PROVIDER), NullPointerException.class);
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(INVALID_STRING, SUNJCE_PROVIDER), NoSuchAlgorithmException.class);
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(KDF_ALGORITHMS[0], (Provider) null), NullPointerException.class);
+
+    // getInstance(String algorithm, KDFParameters kdfParameters)
+    // null spec is a valid case but different class is not
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(null, (KDFParameters) null), NullPointerException.class);
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(INVALID_STRING, (KDFParameters) null),
+        NoSuchAlgorithmException.class);
+    Utils.runAndCheckException(
+        () -> KDF.getInstance(KDF_ALGORITHMS[0], (KDFParameters) new KDFAlgorithmParameterSpec()),
+        ClassCastException.class);
```

Comment:
The ClassCastException is the result of line 297? I don't see the reason for testing this, seems nothing to do with the KDF impls?

Reply:
I agree with @valeriepeng. These test calls can be removed.